### ISSUE TITLE
Windows C++ Project Building

### ISSUE THREAD

**Issue by sseyod** [getsentry/sentry-native#19]
Hi - was hoping to build a test project for C++ / Windows...
However, the steps to follow for configuring sentrypad (and it's dependencies) all seem to be missing from this repo.
For example: I found I first had to install the Ubuntu tool for Windows, and use it to configure the project with premise (having first downloaded and build premake5).
The next question I faced, is that it looks necessary to first to build Crashpad or Breakpad - or both (!) - which is recommended / required?
Basically: it looks like there is a lot of missing info required to actually get going - it'd make sense to me if  sentrypad came pre-built and "Ready To Go" for Windows (and macOS etc.).
I hope this feedback helps...

**Comment by mitsuhiko** [comment]
All of that will become available when this is ready to go. It's still work in progress :)

**Comment by sseyod** [comment]
Sorry, looks like I was getting ahead of myself :) Is there a Beta programme I can join to help you get this working for "real developers" ? :)

**Comment by tonyo** [comment]
@sseyod 

> it'd make sense to me if sentrypad came pre-built and "Ready To Go"

We have plans for distributing the prebuilt packages as well, but turns out, some potential users actually prefer to build crashpad/breakpad themselves (specific build pipelines, exotic target platforms, etc.)

We're currently providing sources + build files for sentrypad and crashpad, for both MacOS and Windows. If you want to try it out, please feel free to download the release package from releases (https://github.com/getsentry/sentrypad/releases/tag/0.0.3), and take a look at the [README](https://github.com/getsentry/sentrypad/blob/master/README.md) for the build instructions.

**Comment by sseyod** [comment]
Hi @tonyo!

I've just dug-in to this; the code you provided seems to build OK with Visual Studio 2019.

However, so far as I can tell (which might be wrong) C++ apps for the Microsoft Store can't include separate binaries - whereas your implementation builds and uses SentryPad as a separate binary (const char *handler_path = "bin/Release/crashpad_handler.exe";)

Have you considered building/using SentryPad as a static library? That way, I guess that C++ apps for the AppStore could use it.

NB the sort of app I'm referring to is documented here:
https://docs.microsoft.com/en-gb/windows/msix/desktop/desktop-to-uwp-packaging-dot-net

Best wishes, Pete

**Comment by mitsuhiko** [comment]
We will add a breakpad backend which is completely static linkable but we did not finish that yet. Crashpad runs out of process so it cannot be statically linked entirely.

**Comment by sseyod** [comment]
Hey, that is *really* great news. Many thanks :) Pete

**Comment by Swatinem** [comment]
@sseyod We are in the process of releasing a complete rewrite later today, which should hopefully make this issue obsolete.

As @mitsuhiko mentioned, crashpad is out-of-process so it has a separate `.exe` by design, so it can’t be integrated. However, sentry-native itself can now be configured and built as a static library.
